### PR TITLE
9C hotfix: Multiplex the order of staged transactions

### PR DESCRIPTION
**Pull request by dahlia**
Co-authored-by: Lee Suho <riemannulus@hitagi.moe>

### CODE REVIEW

File: Libplanet.Tests/Blockchain/BlockChainTest.Internals.cs [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+            PrivateKey c = new PrivateKey();
+            PrivateKey d = new PrivateKey();
+            PrivateKey e = new PrivateKey();
+
+            // 우리가 인위적으로 만들어야 하는 corner cases
+            // A. 정상 케이스 (3 txs: 0, 1, 2)
+            // B. 순서가 뒤집힌 nonce ( 2 tsx: 1, 0)
+            // C. 작은 nonce가 나중 timestamp (2 txs: 0 (later), 1)
+            // D. 이가 빠진 nonce (3 txs: 0, 1, 3)
+            // E. nonce가 겹치는 경우 (4 txs: 0, 1, 1, 2)
```
Comment (limebell) [inline Libplanet.Tests/Blockchain/BlockChainTest.Internals.cs:39]:
```suggestion
            // Corner cases
            // A. Formal case (3 txs: 0, 1, 2)
            // B. Reverse-ordered nonce ( 2 tsx: 1, 0)
            // C. Smaller nonce has later timestamp (2 txs: 0 (later), 1)
            // D. Skipped nonce (3 txs: 0, 1, 3)
            // E. Duplicated nonce (4 txs: 0, 1, 1, 2)
```

Comment (dahlia) [reply Libplanet.Tests/Blockchain/BlockChainTest.Internals.cs:39]: I am going to submit a separate equivalent patch to the 0.10-maintenance branch, and your suggestion will be included there.

**Review by limebell — APPROVED** [review]

File: Libplanet/Blockchain/BlockChain.cs [hunk 2]
```diff
@@ -1556,6 +1635,27 @@ internal BlockLocator GetBlockLocator(int threshold = 10)
...
+            Dictionary<Address, LinkedList<Transaction<T>>> seats = txs
+                .GroupBy(tx => tx.Signer)
+                .Select(g => (g.Key, new LinkedList<Transaction<T>>(g.OrderBy(tx => tx.Nonce))))
+                .ToDictionary(pair => pair.Item1, pair => pair.Item2);
+
+            return txs.Select(tx =>
+            {
+                LinkedList<Transaction<T>> seat = seats[tx.Signer];
+                Transaction<T> first = seat.First.Value;
+                seat.RemoveFirst();
```
Comment (longfin) [inline Libplanet/Blockchain/BlockChain.cs:1654]: Is it needed? (`seat` seems just local variable...)

Comment (longfin) [reply Libplanet/Blockchain/BlockChain.cs:1654]: Ah, I see. `seats[tx.Signer]` is created and referred out side of this block.

**Review by longfin — APPROVED** [review]

**Review by earlbread — APPROVED** [review]
